radio/src/telemetry/mavlink.cpp

```diff
@@ -607,114 +561,28 @@ static inline void MAVLINK_msg_request_data_stream_pack_send(uint8_t req_stream_
 }
 
 //! \brief old mode switch funtion
-static inline void MAVLINK_msg_action_pack_send(uint8_t action) {
-//	mavlink_channel_t chan = MAVLINK_COMM_0;
-//	mavlink_msg_action_send(chan, mavlink_system.sysid, mavlink_system.compid, action);
-}
-
-//! \brief old mode switch funtion
 static inline void MAVLINK_msg_set_mode_send(uint8_t mode) {
 	mavlink_channel_t chan = MAVLINK_COMM_0;
 	mavlink_msg_set_mode_send(chan, mavlink_system.sysid, mode, 0);
 }
 
-/*!	\brief Looks like som kind of task switcher on a timer
- *	\todo Figure out where this was used for and intergrate in current
- *	implemnetation. Funcion dissabled without any side affects.
- */
-#if 0
- void MAVLINK10mspoll(uint8_t count) {
-	switch (count) {
-	case 2: // MAVLINK_MSG_ID_ACTION
-		if (watch_mav_req_id_action > 0) {
-			watch_mav_req_id_action--;
-			// Repeat  is not ack : 150ms*0x07
-			if ((watch_mav_req_id_action & 0x07) == 0) {
-				uint8_t action = MAVLINK_CtrlMode2Action(telemetry_data.req_mode);
-				MAVLINK_msg_action_pack_send(action);
-				char *ptr = mav_statustext;
-				*ptr++ = 'R';
-				*ptr++ = 'Q';
-				*ptr++ = ' ';
-				*ptr++ = action / 10 + '0';
-				*ptr++ = action % 10 + '0';
-				*ptr++ = 0;
-			}
-		}
-		if (telemetry_data.ack_result < 5) {
-			if (telemetry_data.ack_result > 0) {
-				telemetry_data.ack_result++;
-			}
-		}
-		break;
-#ifdef MAVLINK_PARAMS
-	case 4: // MAVLINK_MSG_ID_PARAM_REQUEST_LIST
-		if (watch_mav_req_params_list > 0) {
-			watch_mav_req_params_list--;
-			if (watch_mav_req_params_list == 0) {
-				mav_req_params_nb_recv = 0;
-				MAVLINK_msg_param_request_list_send();
-				watch_mav_req_params_list = 20;
-			}
-		}
-		break;
-#endif
-	case 6: // MAVLINK_MSG_ID_REQUEST_DATA_STREAM
-		if (watch_mav_req_start_data_stream > 0) {
-			watch_mav_req_start_data_stream--;
-			if (watch_mav_req_start_data_stream == 0) {
-				uint8_t req_stream_id = 2;
-				uint16_t req_message_rate = 1;
-
-				MAVLINK_msg_request_data_stream_pack_send(req_stream_id, req_message_rate, data_stream_start_stop);
-				watch_mav_req_start_data_stream = 20;
-				data_stream_start_stop = 1; // maybe start next time
-			}
-		}
-		break;
-#ifdef MAVLINK_PARAMS
-	case 8: // MAVLINK_MSG_ID_PARAM_SET
-		if (watch_mav_req_params_set > 0) {
-			watch_mav_req_params_set--;
-			if (watch_mav_req_params_set == 0) {
-				for (uint8_t idx = 0; idx < NB_PARAMS; idx++) {
-					if (getParam(idx)->repeat) {
-						getParam(idx)->repeat--;
-						MAVLINK_msg_param_set(idx);
-						watch_mav_req_params_set = 3; // 300ms
-						return;
-					}
-				}
-			}
-		}
-		break;
-#endif
-	default:
-		return;
-	}
-}
-#endif
 /*!	\brief Telemetry monitoring, calls \link MAVLINK10mspoll.
  *	\todo Reimplemnt \link MAVLINK10mspoll
  *
  */
 void telemetryWakeup() {
 	uint16_t tmr10ms = get_tmr10ms();
-	uint8_t count = tmr10ms & 0x0f; // 15*10ms
-	if (!count) {
+	static uint16_t last_time = 0;
```

Comment:
Nice catch, this bug is there since we changed the mixer frequency!
I would only keep (and use) the low byte of tmr10ms though.